src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2605,6 +2605,57 @@ class RenameStatementSegment(BaseSegment):
     )
 
 
+@tsql_dialect.segment(replace=True)
+class DropTableStatementSegment(BaseSegment):
+    """A `DROP TABLE` statement.
+
+    Overriding ANSI to add optional delimiter.
+    """
+
+    type = "drop_table_statement"
+    match_grammar = ansi_dialect.get_segment(
+        "DropTableStatementSegment"
+    ).match_grammar.copy(
+        insert=[
+            Ref("DelimiterSegment", optional=True),
+        ],
+    )
```

Comment:
Re-adding these back into T-SQL, as the docstring says, it adds an opitonal delimiter (`;`) to the end of the ANSI statement.